### PR TITLE
US77514 - Lazy load

### PR DESCRIPTION
**Pull request by jstefaniuk-d2l**
- [x] Still needs a loading spinner
- [x] Rebase it

### CODE REVIEW

**Comment by jstefaniuk-d2l** [comment]
Travis fail is due to friday's internet outage

File: d2l-my-courses.html [hunk 1]
```diff
@@ -235,7 +240,7 @@
 					this.$.enrollmentsRootRequest.generateRequest();
 				}
 
-				this.$$('iron-scroll-threshold').scrollTarget = this.$['all-courses'].scrollRegion;
+				this.$$('#all-courses-scroll-threshold').scrollTarget = this.$['all-courses'].scrollRegion;
```
Comment (apalaniuk) [inline d2l-my-courses.html]: Now that `iron-scroll-threshold` is queried by ID and the element is static, we should be able to use [automatic node finding](node-finding) to look it up like on the right-hand side (don't think it's a real issue here, more just a general pattern)

File: d2l-my-courses.html [hunk 2]
```diff
@@ -352,12 +357,17 @@
...
+						this.$.lazyLoadSpinner.scrollIntoView();
 					}
 				}
 			},
 			_setCourseImageEvent: function(e) {
 				this.$$('d2l-all-courses').setCourseImage(e);
 				this.$$('d2l-course-tile-grid').setCourseImage(e);
+			},
+			_changeImageLoadMore: function() {
+				this.$['image-selector-threshold'].scrollTarget = this.$['basic-image-selector-overlay'].scrollRegion;
```
Comment (apalaniuk) [inline d2l-my-courses.html]: Could this be moved into `attached` like the other initialization of a scroll threshold, or does it need to be in here?

File: image-selector/d2l-basic-image-selector.html [hunk 3]
```diff
@@ -94,12 +123,19 @@
...
+					type: String,
+					value: 'hidden'
+				}
 			},
 			behaviors: [ window.D2L.MyCourses.LocalizeBehavior ],
 			_sirenParser: document.createElement('d2l-siren-parser'),
 			_searchImages: [],
 			_defaultImages: [],
-			_displaySearchResults: function(userSearchText) {
+			_scrollThreshold: null,
```
Comment (apalaniuk) [inline image-selector/d2l-basic-image-selector.html]: I think this is unused now

**Comment by apalaniuk** [comment]
LGTM!

**Comment by apalaniuk** [comment]
LGTM. Works as expected on Chrome 53, FF46, IE11 on Win8.1, Edge 14 in Win10 VM, Safari 9.1.

**Pull request merged**